opentelemetry-api/src/opentelemetry/propagation/__init__.py

```diff
@@ -0,0 +1,144 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+
+import opentelemetry.context.propagation.httptextformat as httptextformat
+import opentelemetry.trace as trace
+from opentelemetry.context import Context
+from opentelemetry.context.propagation import (
+    ContextT,
+    DefaultHTTPExtractor,
+    DefaultHTTPInjector,
+    Getter,
+    Setter,
+)
+
+
+def extract(
+    carrier: ContextT,
+    context: typing.Optional[Context] = None,
+    extractors: typing.Optional[
+        typing.List[httptextformat.HTTPExtractor]
+    ] = None,
+    get_from_carrier: typing.Optional[Getter[ContextT]] = None,
+) -> typing.Optional[Context]:
+    """Load the parent SpanContext from values in the carrier.
+
+    Using the specified HTTPExtractor, the propagator will
+    extract a SpanContext from the carrier. If one is found,
+    it will be set as the parent context of the current span.
+
+    Args:
+        get_from_carrier: a function that can retrieve zero
+            or more values from the carrier. In the case that
+            the value does not exist, return an empty list.
+        carrier: and object which contains values that are
+            used to construct a SpanContext. This object
+            must be paired with an appropriate get_from_carrier
+            which understands how to extract a value from it.
+    """
+    if context is None:
+        context = Context.current()
+    if extractors is None:
+        extractors = get_http_extractors()
+
+    for extractor in extractors:
+        # TODO: improve this
+        if get_from_carrier:
+            return extractor.extract(
+                context=context,
+                carrier=carrier,
+                get_from_carrier=get_from_carrier,
+            )
+        return extractor.extract(context=context, carrier=carrier)
+
+    return None
+
+
+def inject(
+    carrier: ContextT,
+    injectors: typing.Optional[
+        typing.List[httptextformat.HTTPInjector]
+    ] = None,
+    context: typing.Optional[Context] = None,
+) -> None:
+    """Inject values from the current context into the carrier.
+
+    inject enables the propagation of values into HTTP clients or
+    other objects which perform an HTTP request. Implementations
+    should use the set_in_carrier method to set values on the
+    carrier.
+
+    Args:
+        set_in_carrier: A setter function that can set values
+            on the carrier.
+        carrier: An object that contains a representation of HTTP
+            headers. Should be paired with set_in_carrier, which
+            should know how to set header values on the carrier.
+    """
+    if context is None:
+        context = Context.current()
+    if injectors is None:
+        injectors = get_http_injectors()
+
+    for injector in injectors:
+        injector.inject(context=context, carrier=carrier)
+
+
+_HTTP_TEXT_INJECTORS = [
+    DefaultHTTPInjector
```

Comment:
If these are only used in `inject` and `extract`, what's the difference between an empty list and a list with a single no-op injector/extractor?